find/main/routes.py

```diff
@@ -175,6 +178,49 @@ def retrieve_download(filename: str):
         return render_template("find/main/retrieve-download.html", context=context, form=form)
 
 
+@bp.route("/retrieve-spreadsheet/<fund_code>/<submission_id>", methods=["GET", "POST"])
+@login_required(return_app=SupportedApp.POST_AWARD_FRONTEND)
+@check_internal_user
+def retrieve_spreadsheet(fund_code: str, submission_id: str):
+    object_name = f"{fund_code}/{submission_id}"
+
+    try:
+        file_metadata = get_file_metadata(Config.AWS_S3_BUCKET_SUCCESSFUL_FILES, object_name, True)
+    except FileNotFoundError:
+        if request.method == "POST":
+            return redirect(
+                url_for(
+                    ".retrieve_spreadsheet",
+                    fund_code=fund_code,
+                    submission_id=submission_id,
+                )
+            )
+
+        return render_template("find/main/spreadsheet-not-found.html")
```

Comment:
I think we can probably just render a normal 404 page - do we need a custom page for this? The spreadsheets don't have a short TTL on them, unlike the download reports, so this feels less likely to happen.

Reply:
:+1:

Reply:
Returning the common `404.html` page now.